### PR TITLE
deps: support Kubernetes 1.28

### PR DESCRIPTION
**Pull request by 3u13r**
### Proposed change(s)

- Remove K8s Version 1.25
- Make 1.27 the default version
- Add support for 1.28

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

**Comment by 3u13r** [comment]
Azure, 1.28, sonobuoy quick: https://github.com/edgelesssys/constellation/actions/runs/5877406153
GCP, 1.28, autoscaling: https://github.com/edgelesssys/constellation/actions/runs/5877415773
AWS, 1.28, lb: https://github.com/edgelesssys/constellation/actions/runs/5877422771

File: internal/versions/versions.go [hunk 1]
```diff
@@ -104,9 +104,9 @@ const (
 	//
 
 	// KonnectivityAgentImage agent image for konnectivity service.
-	KonnectivityAgentImage = "registry.k8s.io/kas-network-proxy/proxy-agent:v0.1.2@sha256:cd3046d253d26ffb5907c625e0d0c2be05c5693c90e12116980851739fc0ead8" // renovate:container
+	KonnectivityAgentImage = "registry.k8s.io/kas-network-proxy/proxy-agent:v0.1.3@sha256:c6c222e2e56bd333a1d048861a29bd226c3dbfb732d3dfc0178118da3f581575" // renovate:container
```
Comment (katexochen) [inline internal/versions/versions.go:107]: Please do this in a separate PR.

File: internal/versions/versions.go [hunk 2]
```diff
@@ -283,7 +283,7 @@ var VersionConfigs = map[ValidK8sVersion]KubernetesVersion{
 		// TODO(3u13r): use newer "cloud-provider-gcp" from https://github.com/kubernetes/cloud-provider-gcp when newer releases are available.
 		CloudControllerManagerImageGCP: "ghcr.io/edgelesssys/cloud-provider-gcp:v26.0.1@sha256:db2b15a20ad690784a6015bfad55c4dff15826be8cf9f6ac77d70abd11b1f70c", // renovate:container
 		// CloudControllerManagerImageOpenStack is the CCM image used on OpenStack.
-		CloudControllerManagerImageOpenStack: "docker.io/k8scloudprovider/openstack-cloud-controller-manager:v1.26.2", // renovate:container
+		CloudControllerManagerImageOpenStack: "docker.io/k8scloudprovider/openstack-cloud-controller-manager:v1.26.3@sha256:65f0945ea9fc17e64812befbf3fc52b06c13df1c3407cb8022e8110a2fe08a4a", // renovate:container
```
Comment (katexochen) [inline internal/versions/versions.go:286]: Could you upgrade the other instances of openstack ccm as well?

**Review by katexochen — APPROVED** [review]

**Review by thomasten — APPROVED** [review]
